lib/parse/client.rb

```diff
@@ -69,6 +69,11 @@ def request(uri, method = :get, body = nil, query = nil, content_type = nil)
       end
 
       @session.send(method, uri, query || body || {}, headers).body
+
+    # Don't leak our internal libraries to our clients.
+    # Extend this list of exceptions as needed.
+    rescue Faraday::Error::ConnectionFailed, Faraday::Error::TimeoutError
+      raise Parse::ConnectionError
```

Comment:
You are correct. Probably the patch would be bigger because the exception objects have to take into account the wrapped exception.

The thing I'm worried about is that all the information about what cause the exceptions you're rescuing is going to get lost.

We could go about it like this:

- I approve your patch as it is
- you open a bug about the exception problem (and add a reference to this PR)
- before 1.0 I do a review of all the exception handling and pass along wrapped exceptions.

How does it sound?